package.json

```diff
@@ -20,30 +20,30 @@
   "keywords": [
     "actions",
     "node",
-    "setup"
+    "setup",
+    "github",
+    "teams"
   ],
   "author": "Richard Macklin",
   "license": "MIT",
   "dependencies": {
-    "@actions/core": "^1.2.0",
-    "@actions/github": "^2.1.1",
-    "@octokit/rest": "^16.43.1",
+    "@actions/core": "^1.4.0",
+    "@actions/github": "^5.0.0",
```

Comment:
The update from `2.1.1` to `5.0.0` is really critical for me since it doesn't work on Github enterprise without that:

https://github.com/actions/toolkit/blob/main/packages/github/RELEASES.md#220

Reply:
Got it - thanks for explaining that! Based on the changelog link you provided, it seems like upgrading to 2.2.0 is sufficient, so I want to start there without switching to a new major version yet (since the major version upgrades could include breaking changes). I've opened a new PR for that upgrade: https://github.com/rmacklin/team-sync/pull/14.

Can you give it a try with
```yaml
    steps:
    - uses: rmacklin/team-sync@5617c3b5dd662e0b0b35f93bbf4cb46b416c7b0b
```
in your workflow definition?

Reply:
Got it - thanks for explaining that! Based on the changelog link you provided, it seems like upgrading to 2.2.0 is sufficient, so I want to start there without switching to a new major version yet (since the major version upgrades could include breaking changes). I've opened a new PR for that upgrade: https://github.com/rmacklin/team-sync/pull/14.

Can you give it a try with
```yaml
    steps:
    - uses: rmacklin/team-sync@5617c3b5dd662e0b0b35f93bbf4cb46b416c7b0b
```
in your workflow definition, and let me know if it works on your GitHub Enterprise Server?

Reply:
I'm already running it on my `master` branch (which is subject of this PR). So no need for an intermediate step.

I face however a very strange timing issue. I try to sync ~20 teams. but it always fails after the ~15th team with the following error:

`Error: HttpError: request to https://<GHES-API>/teams failed, reason: connect ETIMEDOUT`
Not sure how to proceed. Especially since it's working 14 times and then fails...

Reply:
Okay, I've merged #14.

I haven't seen that error personally, but I don't use GitHub Enterprise. Using the action with an organization on github.com, I've seen it sync over 40 teams successfully. It seems possible that a self-hosted GitHub Enterprise deployment may not be able to handle the volume of successive requests to query for team memberships and then add/remove team members accordingly. That's just a guess, though, so I'd suggest cross referencing any monitoring you have in place for the GitHub Enterprise deployment to look for indications that it's timing out because it's unable to handle the load.

Reply:
I just tried `rmacklin/team-sync@main`: Doesn' t work:

![image](https://user-images.githubusercontent.com/448282/129006061-6319e640-741f-412d-aa34-c546fdc7218a.png)

Reply:
Ok, I think it might be necessary to upgrade to `@actions/github` to v3.0.0 to pull in https://github.com/actions/toolkit/commit/4a89cf72de68713d3a465254e90537d8ea5b421e (upgrading further includes [breaking changes to Octokit](https://github.com/octokit/plugin-rest-endpoint-methods.js/releases/tag/v4.0.0), which I'd like to handle separately).

I've upgraded to v3.0.0 here: https://github.com/rmacklin/team-sync/commit/042d3b844823ea5ae2a03c23ea55e1c822deab51. If you update your mirror, you can try again with
`uses: ghcom-actions/rmacklin-team-sync@042d3b844823ea5ae2a03c23ea55e1c822deab51`.

(I do still suspect that it's more likely an issue with the GitHub Enterprise deployment being overloaded, but this may at least get past the "Bad credentials" error.)

Reply:
Must be something on the runner. When running locally it worked on my `MichaelSp/team-sync@master` branch.
Haven't checked your branch.

Sorry, but I have to say: I don't see a reason to break this down into pieces. 
Why:
 * If you write the tests first, they would be for the old API, so you gonna write them twice.
 * If you do the upgrade first, you could break things, since you have no tests...
 
Conclusion: Happy with what I have now.

Digging into the runner now, to see what is wrong there....